diff --git a/linux-user/syscall.c b/linux-user/syscall.c
index f09f676..6d52704 100644
--- a/linux-user/syscall.c
+++ b/linux-user/syscall.c
@@ -6669,7 +6669,8 @@ static inline abi_long target_to_host_sigevent(struct sigevent *host_sevp,
     host_sevp->sigev_signo =
         target_to_host_signal(tswap32(target_sevp->sigev_signo));
     host_sevp->sigev_notify = tswap32(target_sevp->sigev_notify);
-    host_sevp->_sigev_un._tid = tswap32(target_sevp->_sigev_un._tid);
+    // host_sevp->_sigev_un._tid = tswap32(target_sevp->_sigev_un._tid);
+    // not present in struct sigevent on Alpine Linux 3.4
 
     unlock_user_struct(target_sevp, target_addr, 1);
     return 0;
